### ISSUE TITLE
Error compiling CUDA kernels

### ISSUE THREAD

**Issue by pgrinaway** [openmm/openmm#1879]
Hi,

(system, state, and integrator xmls attached). When I run this code:

```python
import simtk.openmm as openmm
import simtk.openmm.app as app
import simtk.unit as unit

sys_file = open('system.xml', 'r')
sys_string = "".join(sys_file.readlines())
sys_file.close()

integrator_file = open('integrator.xml','r')
integrator_string = "".join(integrator_file.readlines())
integrator_file.close()

state_file = open('state.xml','r')
state_string = "".join(state_file.readlines())
state_file.close()

system = openmm.XmlSerializer.deserialize(sys_string)
integrator = openmm.XmlSerializer.deserialize(integrator_string)
state = openmm.XmlSerializer.deserialize(state_string)

platform = openmm.Platform.getPlatformByName("OpenCL")

ctx = openmm.Context(system, integrator, platform)
ctx.setState(state)
print(ctx.getState(getEnergy=True).getPotentialEnergy())
```

I get this error:

```
/home/pgrinaway/anaconda3/lib/python3.6/site-packages/simtk/openmm/openmm.py in getState(self, getPositions, getVelocities, getForces, getEnergy, getParameters, getParameterDerivatives, enforcePeriodicBox, groups)
  11873         if getParameterDerivatives:
  11874             types += State.ParameterDerivatives
> 11875         state = _openmm.Context_getState(self, types, enforcePeriodicBox, groups_mask)
  11876         return state
  11877 

Exception: Error compiling kernel: ptxas error   : Entry function 'findBlocksWithInteractions' uses too much shared data (0x10390 bytes, 0xc000 max)
```

Alternatively, if I switch from using the OpenCL platform to the CUDA platform, I get this error:

```
/home/pgrinaway/anaconda3/lib/python3.6/site-packages/simtk/openmm/openmm.py in getState(self, getPositions, getVelocities, getForces, getEnergy, getParameters, getParameterDerivatives, enforcePeriodicBox, groups)
  11873         if getParameterDerivatives:
  11874             types += State.ParameterDerivatives
> 11875         state = _openmm.Context_getState(self, types, enforcePeriodicBox, groups_mask)
  11876         return state
  11877 

Exception: Error loading CUDA module: CUDA_ERROR_NO_BINARY_FOR_GPU (209)
```

Some additional details: I'm running CUDA 8.0, GeForce GTX 1080, driver version 367.48

[xmlfiles.zip](https://github.com/pandegroup/openmm/files/1245737/xmlfiles.zip)

The commit hash of OpenMM is ostensibly the latest one in `conda`, `18a7cac0e14789756429d4dd6418553adc36c844`. I can try to compile the latest commit too, will report back. 

If there's anything else that I could do to help, please feel free to ask. Thanks!

**Comment by pgrinaway** [comment]
Apparently, if I remove the `NonbondedForce`, it actually succeeds in giving me a potential energy.

**Comment by pgrinaway** [comment]
Also, re-trying with release conda version commit hash `c1a64aaa3b4b71f8dd9648fa724d2548a99d4ced` (though this failed before as well)

**Comment by pgrinaway** [comment]
The release version does fail with the same error.

**Comment by pgrinaway** [comment]
Looks like the issue has to do with the number of exceptions (ca. 750,000). I assume these are somehow stored in shared memory, and so the number one can add is quite limited?

**Comment by peastman** [comment]
I'm getting an error trying to download your file:

> The request signature we calculated does not match the signature you provided. Check your key and signing method.

**Comment by pgrinaway** [comment]
Huh, that's weird. Let me try again--they download fine for me.

[xmlfiles.zip](https://github.com/pandegroup/openmm/files/1249462/xmlfiles.zip)

**Comment by peastman** [comment]
Ok, I managed to download it.  I think I understand what's going on with the OpenCL platform.  You have a bunch of different nonbonded forces with a whole lot of per-particle parameters between them.  All those forces get combined into a single kernel, and the per-particle parameters get cached in shared memory.  There isn't enough for all of them.

But the CUDA platform shouldn't have that problem.  Running on any recent GPU, it will use shuffle operations to pass per-particle parameters between threads rather than storing them in shared memory.  Perhaps it's running out of space for variables?  But even then, it should just offload them to global memory, which hurts performance but should still work.  I'll investigate to figure out what's going on.

**Comment by peastman** [comment]
By the way, your code for loading the XML files can be a lot simpler:

```Python
with open('system.xml') as input:
  system = XmlSerializer.deserialize(input.read())
```

etc.
